cmd/list.go

```diff
@@ -181,27 +183,58 @@ var listGroupProjectsCmd = &cobra.Command{
 	},
 }
 
-var listProjectCmd = &cobra.Command{
+var listEnvironmentsCmd = &cobra.Command{
 	Use:     "environments",
 	Aliases: []string{"e"},
 	Short:   "List environments for a project (alias: e)",
-	Run: func(cmd *cobra.Command, args []string) {
+	PreRunE: func(_ *cobra.Command, _ []string) error {
+		return validateTokenE(lagoonCLIConfig.Current)
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		debug, err := cmd.Flags().GetBool("debug")
+		if err != nil {
+			return err
+		}
 		if cmdProjectName == "" {
 			fmt.Println("Missing arguments: Project name is not defined")
 			cmd.Help()
 			os.Exit(1)
 		}
-		returnedJSON, err := pClient.ListEnvironmentsForProject(cmdProjectName)
-		handleError(err)
-		var dataMain output.Table
-		err = json.Unmarshal([]byte(returnedJSON), &dataMain)
-		handleError(err)
-		if len(dataMain.Data) == 0 {
-			output.RenderInfo(fmt.Sprintf("There are no environments for project '%s'", cmdProjectName), outputOptions)
-			os.Exit(0)
+
+		current := lagoonCLIConfig.Current
+		token := lagoonCLIConfig.Lagoons[current].Token
+		lc := lclient.New(
+			lagoonCLIConfig.Lagoons[current].GraphQL,
+			lagoonCLIVersion,
+			&token,
+			debug)
+		environments, err := l.GetEnvironmentsByProjectName(context.TODO(), cmdProjectName, lc)
+		if err != nil {
+			return err
 		}
-		output.RenderOutput(dataMain, outputOptions)
 
+		data := []output.Data{}
+		for _, environment := range *environments {
+			var envRoute = "none"
+			if environment.Route != "" {
+				envRoute = environment.Route
+			}
+			data = append(data, []string{
+				returnNonEmptyString(fmt.Sprintf("%d", environment.ID)),
+				returnNonEmptyString(fmt.Sprintf("%v", environment.Name)),
+				returnNonEmptyString(fmt.Sprintf("%v", environment.DeployType)),
+				returnNonEmptyString(fmt.Sprintf("%v", environment.EnvironmentType)),
+				returnNonEmptyString(fmt.Sprintf("%v", environment.OpenshiftProjectName)),
+				returnNonEmptyString(fmt.Sprintf("%v", envRoute)),
+				returnNonEmptyString(fmt.Sprintf("%v", environment.DeployTarget.Name)),
+			})
+		}
+		dataMain := output.Table{
+			Header: []string{"ID", "Name", "DeployType", "Environment", "OpenshiftProjectName", "Route", "DeployTarget"},
```

Comment:
I think we should change this from `OpenshiftProjectName` to `Namespace`, what do you think?

Reply:
Yep I like it, much better!